### PR TITLE
queryRenderedFeatures: default params to empty object instead of throwing error

### LINKED ISSUE #2969: Fix bug preventing "queryRenderedFeatures" from being called with no arguments
The GL JS API documentation correctly lists this method's signature as 

```js
queryRenderedFeatures([region], [parameters])
```

... however calling `map.queryRenderedFeatures()` produces an error unless a parameters object is passed `map.queryRenderedFeatures({})`. 

This bug was causing some of our benchmark scripts to crash and was fixed in https://github.com/mapbox/mapbox-gl-js/pull/2950 ([exact spot in the diff](https://github.com/mapbox/mapbox-gl-js/pull/2950/files#diff-63abd99e8332ce21c1323063948b72f0R39))

Relevant source [here](http://github.com/mapbox/mapbox-gl-js/blob/f287983ecf88ee4df6ad0ee16144de02d93477ae/js/ui/map.js#L516-L517)

### PR DESCRIPTION
**Pull request by mapsam**
#### What changed

Added a default empty object `{}` to `map.queryRenderedFeatures()` if nothing is passed through. Currently gl-js throws an error if the query is missing a second parameter. In order to pass linting tests, I had to rename `params` to `parameters` within the logic of `queryRenderedFeatures`.

fixes #2969

#### Tests

Added two new tests that confirm the object is generated if it isn't passed through, and a test that confirms all parameters are preserved in the output (happy to remove this if it's unnecessary).

#### Refs

#2647 

cc @lucaswoj @bhousel @mollymerp

### CODE REVIEW

File: test/js/ui/map.test.js [hunk 1]
```diff
@@ -593,6 +593,26 @@ test('Map', function(t) {
...
+                map.style.queryRenderedFeatures = function (coords, o) {
+                    t.ok(coords);
+                    t.deepEqual(o, {foo: 'bar'}, 'options are expected');
+                    t.end();
+                };
+
+                map.queryRenderedFeatures(map.project(new LngLat(0, 0)), {foo: 'bar'});
+            });
+
             t.end();
```
Comment (lucaswoj) [inline test/js/ui/map.test.js]: The shared `map` across these test cases make me a little nervous but I don't see any immediate harm coming of it.

File: test/js/ui/map.test.js [hunk 2]
```diff
@@ -593,6 +593,26 @@ test('Map', function(t) {
...
+
+            t.test('options carry through query', function(t) {
+                map.style.queryRenderedFeatures = function (coords, o) {
+                    t.ok(coords);
+                    t.deepEqual(o, {foo: 'bar'}, 'options are expected');
+                    t.end();
+                };
+
+                map.queryRenderedFeatures(map.project(new LngLat(0, 0)), {foo: 'bar'});
+            });
```
Comment (lucaswoj) [inline test/js/ui/map.test.js]: In the original issue I was getting exceptions from calling `map.queryRenderedFeatures()` (no arguments). Can we add another test for the "no arguments" case?

File: test/js/ui/map.test.js (hunk 1 above)
Comment (mapsam) [inline test/js/ui/map.test.js]: @lucaswoj I grabbed this from the [two](https://github.com/mapbox/mapbox-gl-js/pull/2979/files/76381412754bc117f0c6d35f030c50c49694e0cb#diff-277bb5e9f409aee234cc524db362a55bR568) tests [above](https://github.com/mapbox/mapbox-gl-js/pull/2979/files/76381412754bc117f0c6d35f030c50c49694e0cb#diff-277bb5e9f409aee234cc524db362a55bR580) without thinking much about it. Happy to find an alternative!

Comment (lucaswoj) [inline test/js/ui/map.test.js]: Totally understood! I appreciate you sticking to convention. 😄 Let's leave these tests-as is for now.

File: test/js/ui/map.test.js (hunk 2 above)
Comment (mapsam) [inline test/js/ui/map.test.js]: Oh! Totally.

Comment (mapsam) [inline test/js/ui/map.test.js]: @lucaswoj added an extra default for `pointOrBox` as well (defaults to `undefined`, matching the syntax below).

File: js/style/style.js [hunk 3]
```diff
@@ -644,7 +644,7 @@ Style.prototype = util.inherit(Evented, {
         }
 
         var includedSources = {};
-        if (params.layers) {
+        if (params && params.layers) {
```
Comment (mapsam) [inline js/style/style.js]: @lucaswoj In the test [here](https://github.com/mapbox/mapbox-gl-js/pull/2979/files#diff-277bb5e9f409aee234cc524db362a55bR567) I hit the error `Error: TypeError: Cannot read property 'layers' of undefined`, which led to a check to ensure `params` exists in the first place.

**Pull request merged**